server/services/omnichannel-voip/service.ts

```diff
@@ -17,12 +23,71 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 
 	private users: UsersRaw;
 
+	private voipRoom: VoipRoomsRaw;
+
 	constructor(db: Db) {
 		super();
 		this.users = new UsersRaw(db.collection('users'));
+		this.voipRoom = new VoipRoomsRaw(db.collection('rocketchat_room'));
 		this.logger = new Logger('OmnichannelVoipService');
 	}
 
+	private normalizeAgent(agentId: string): Record<string, unknown> | undefined {
+		if (!agentId) {
+			return;
+		}
+
+		if (!settings.get('Livechat_show_agent_info')) {
+			return { hiddenInfo: true };
+		}
+
+		const agent = Users.getAgentInfo(agentId);
+		const { customFields: agentCustomFields, ...extraData } = agent;
+
+		return Object.assign(extraData);
+	}
+
+	private async createVoipRoom(rid: string, name: string, agent: any, guest: ILivechatVisitor): Promise<string> {
```

Comment:
agent is an object with `agentId` and `username` attributes

Reply:
Fixed.